### ISSUE TITLE
(cbor) exception for major type 4 not supported on a particular payload

### ISSUE THREAD

**Issue by axelczk** [FasterXML/jackson-dataformats-binary#280]
I was doing some test on my projet using your library, and an exception was thrown for a major type 4 not supported on this payload:

Diagnostic payload: {[3,0,0]:"test"}

But the major type 4 is working with this one: [{0: "0"}, {0: "1"}]

Is this a problem on the library ? Or it's not supported ? In the second case, do you plan to support it in the future ?

**Comment by cowtowncoder** [comment]
@axelczk Please include some kind of reproduction, and perhaps stack trace / exception message. Is the issue with values or key type? And why are you using `readValues()` (or should that be `readValue()`)?

**Comment by axelczk** [comment]
Here is a payload that fails:

```
0xA1, 0x83 , 0x03, 0x00, 0x00, 0x64, 0x74, 0x65, 0x73, 0x74 
```

The problem is: the library doesn't accept array as a fieldName. 

The exception is thrown in: _decodeNonStringName(int) : 
```
throw _constructError("Unsupported major type ("+type+") for CBOR Objects, not (yet?) supported, only Strings");
```

**Comment by cowtowncoder** [comment]
Correct: this is not supported and I am not quite sure how it could be -- it would require transformation of some kind from whatever array type is being used into Java `String`, essentially.
If this was, say, a `byte[]` equivalent containing UTF-8 encoded name, that'd be easy enough, but without some kind of metadata or configuration it is not safe to assume this in general.

**Comment by axelczk** [comment]
I would say you should deserialize it like an array value but it's easier said than done. 

Thanks you for your answer. It's very appreciated !

**Comment by cowtowncoder** [comment]
The problem is not deserialization as these values can be decoded as values. The problem is that of exposing it via token stream; Object keys are exposed as `JsonToken.FIELD_NAME` for which `String` accessor (only) exists. Plus some support for `int`/`long` keys.
While it'd be possible to expose other types for CBOR backend, the question beyond that is how to make use of that via databind.

So it is all possible, just quite a bit of work and needing to fit with the API approach.

**Comment by fbacchella** [comment]
As the current implementation of CBOR don’t handle custom tags and will probably not do it in short to medium delay, I gave up using it and use directly a `com.fasterxml.jackson.dataformat.cbor.CBORParser`.

In this context, I could handle directly custom key type, but as the parser only expose it a a String, I’m stuck. Just adding direct access of the effective real value would be helpfull.

It’s very frustrating, as the encoding is done right: `Map.of(65535, 2)` is serialized as `BF 19 FF FF 02 FF`.

And for serialization, that should be a feature flag, to be able to send it to remote peer that can handle it.
